keps/sig-api-machinery/1040-priority-and-fairness/README.md

```diff
@@ -650,450 +650,484 @@ where SCL is the apiserver's concurrency limit and ACS(l) is the
-duration of a request.  Instead, we use a guess `G`.  If and when the
-guess turns out to be too short --- that is, its expected end time
-arrives in the virtual world but the request has not finished in the
-real world --- the guess is increased.  Remember that the virtual
-world never serves a request faster than the real world, so whenever
-that adjustment is made we are sure that the guess really was too
-short.
-
-Essentially always the (eventually adjusted, as necessary) guess will
-turn out to be too long.  When the request finishes execution in the
-real world, we learn its actual service duration `D`.  The completion
-in the virtual world is concurrent or in the future, never the past.
-At this point we adjust the expected end time of the request in the
-virtual world to be based on `D` rather than the guess.
-
-When the request finishes execution in the virtual world --- which by
-this time is an accurate reflection of the true service duration `D`
---- either another request is dispatched from the same queue or all
-the remaining requests in that queue start getting faster service.  In
-both cases, the service delivery in the virtual world has reacted
-properly to the true service duration.
+Integral[from tau=t_dispatch_virtual(i,j)
+         to   tau=  t_finish_virtual(i,j)] mu_equal(tau) dtau  =  width(i,j) * G
+```
+
+This means that when making real world dispatching decisions, the only
+distinctions among waiting requests are their width estimates (plus
+the previous history of their queue).  Originally we set `G` to a
+value that is normally a gross over-estimate: one minute.  That meant
+that differences in width estimates made grossly inordinate
+differences in dispatching order.  Consider the example where all
+requests actually take 100 ms to execute, and almost all have a width
+of 1.  When a queue has a request of width 2 at its head, that queue's
+estimated next completion time has a penalty compared to all the other
+queues that is equal to the time it takes to execute 600 requests.
+And once that width=2 request eventually is run and completes, that
+queue's following estimated completion times jump forward by 1198
+request's worth of time (as opposed to 599 in the usual case).
+
+If requests all had the same width, we could set `G` to any value and
+get the same behavior as any other value of `G`.  As far as
+dispatching `width=1` requests is concerned, the problem at any time
+is to pick a queue `i` for which the value of
+`t_dispatch_virtual(i,oldest_waiting_j(i)) + G` is minimal.  Clearly,
+the value of `G` does not matter to that ordering among queues.
+
+To remove the inordinate impact of width estimates, we have changed
+`G` to zero.  In other words, the differentiation among queues comes
+only from requests that have completed in the real world.
+
+If we wanted the waiting requests to have _some_ effect, we could set
+`G` to a small amount of time; 100 ms would not be bad.  Additionally,
```

Comment:
nit: IIRC, we were chatting about lower values, order of magnitude of 10ms last time.

Eventually, we may start estimating them better, but that's out-of-scope for this KEP update imho.

Reply:
Updated that part in the force-push to 1a58c3eb